snapcraft/internal/sources.py

```diff
@@ -80,6 +82,28 @@ def __init__(self, source, source_dir, source_tag=None,
         self.source_branch = source_branch
 
 
+class FileBase(Base):
+
+    def pull(self):
+        if common.isurl(self.source):
+            self.download()
+        else:
+            shutil.copy2(self.source, self.source_dir)
+
+        self.provision(self.source_dir)
+
+    def download(self):
+        req = requests.get(self.source, stream=True, allow_redirects=True)
+        if req.status_code is not 200:
+            raise EnvironmentError('unexpected http status code when '
+                                   'downloading {}'.format(req.status_code))
+
+        file = os.path.join(self.source_dir, os.path.basename(self.source))
+        with open(file, 'wb') as f:
+            for chunk in req.iter_content(1024):
+                f.write(chunk)
```

Comment:
Not for this PR, but we really need a progress bar here. Perfect use-case for one.